### DISCUSSION TITLE
Problem z bazą SQL

### DISCUSSION THREAD

**Discussion by xluciola** [AzureCommunityPL/group#136] [Pomoc]
założyłem na Azure serwer i bazę SQL.
do bazy próbuję podłączyć zewnętrzny program ERP
przy podłączeniu do bazy wyskakuje błąd programu , że "do pracy programu wymagane jest uprawnienie VIEW SERVER STATE.
w SQL Sever Management Studio próbuję wpisać GRANT VIEW SERVER STATE TO "USER" i wyskakuje komunikat "securable class "server " not supported in this version of SQL server.
jak mam rozwiązać problem?

**Comment by kamil-mrzyglod** [comment]
Ale założyłeś serwer i bazę jako Azure SQL czy postawiłeś VM z SQL Server?

**Reply by xluciola** [reply]
Założyłem serwer i bazę jako Azure SQL.
na VM nie stawiałem

**Comment by mgrabarz** [comment]
Zamiast Azure SQL potrzebujesz Azure SQL Managed Instance: https://azure.microsoft.com/en-us/services/azure-sql/sql-managed-instance/

**Comment by xluciola** [comment]
czyli w przypadku serwera AZure SQL nie ma możliwości przyznania takich uprawnień i muszę serwer założyć na maszynie wirtualnej i skonfigurować połączenie?

**Reply by mgrabarz** [reply]
W Azure sa 3 opcje:
- postaw sam na VM - to zadziała, ale sam musisz stawiać.
- Azure SQL - nie zadziała, jak sam sprawdziłeś.
- Azure SQL Managed Instance - też zadziała. Trochę jak VM, ale Azure sam stawia i zarządza.

**Comment by xluciola** [comment]
dziękuję za pomoc .
próbuję założyć Azure Sql instance i pojawia się taki komunikat:
Wystąpienie zarządzane nie jest obecnie dostępne dla wybranej subskrypcji i regionu
żaden region nie działa.
czy brakuje zasobów czy moja testowa subskrypcja nie obejmuje tego typu zasobów?

**Reply by mgrabarz** [reply]
Jeśli to free trial to tak może być. 
Oczywiście zawsze możesz poprosić o zwiększenie limitów, ale mogą prośbę odrzucić lub zmieni się trial w płatną.

**Comment by xluciola** [comment]
i kolejny problem podczas tworzenia maszyny wirtualnej sql 👍 
Zasób maszyny wirtualnej SQL ma nieprawidłowy stan na potrzeby zarządzania. Przyczyną może być zły stan maszyny wirtualnej SQL albo nieprawidłowy stan podstawowej maszyny wirtualnej (np. Zatrzymana, Cofnięto przydział lub Niepowodzenie) bądź nie można znaleźć maszyny wirtualnej. Upewnij się, że maszyna wirtualna SQL i podstawowa maszyna wirtualna są w dobrej kondycji.
jak rozwiązac ten problem
dziękuję

**Reply by kamil-mrzyglod** [reply]
Zmień w pierwszej kolejności język portalu na angielski 😉

**Reply by mgrabarz** [reply]
To sprawdź "stan" maszyny. Czyżby nie była "running"?
Generalnie ciężko się podłączyć do wyłączonej maszyny ;)

**Reply by xluciola** [reply]
maszyna jest online
SQL Vitrtual Machine jest offline

**Reply by mgrabarz** [reply]
To ja nie wiem czym jest "SQL Virtual Machine". 🤷
